automation/gtm_automation.js

```diff
@@ -0,0 +1,496 @@
+/*
+  Copyright 2023 Google LLC
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+// The OAuth token client used to authenticate API calls.
+let tokenClient;
+
+// the GTM variables we will create. This is augmented with 'attribution' if
+// the attribution build is used.
+const VariableNames = [
```

Comment:
nit: Use CONSTANT_CASE for global constant values.
See https://google.github.io/styleguide/tsguide.html#naming-rules-by-identifier-type
```suggestion
const VARIABLE_NAMES = [
```